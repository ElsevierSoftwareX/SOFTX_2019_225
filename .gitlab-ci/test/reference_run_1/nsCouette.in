&parameters_grid
m_r   = 32                 ! N radial points           => m_r      grid points (radial)
m_th  = 384                ! M azimuthal Fourier modes => 2*m_th+1 grid points (azimuthal)
m_z0  = 320                ! L axial Fourier modes     => 2*m_z0+1 grid points (axial)
k_th0 = 1.0d0              ! azimuthal wavenumber      => L_th = 2*pi/k_th0 azimuthal length of grid
k_z0  = 3.14159265358979d0 ! axial wavenumber          => L_z  = 2*pi/k_z0  = 7.48 axial length of grid
eta   = 0.500d0            ! aspect ratio => r_i = eta/(1-eta),r_o = 1/(1-eta) inner/outer radius
/

&parameters_physics
Re_i = 700.0d0           ! inner cylinder Reynolds number
Re_o =-700.0d0           ! outer cylinder Reynolds number
/

&parameters_timestep
numsteps    = 100       ! number of computational time steps 
init_dt     = 1.0d-4    ! initial size of time step
variable_dt = T         ! use a variable (=T) or fixed (=F) time step size
maxdt       = 1.0d-2    ! maximum size of time step
Courant     = 0.25      ! CFL safety factor
/

&parameters_output
fBase_ic = 'REF_1'      ! identifier for coeff_ (checkpoint) and fields_ (hdf5) files 
dn_coeff  = 100         ! output interval [steps] for coeff (dn_coeff = -1 disables ouput)
dn_ke     = 1           ! output interval [steps] for energy
dn_vel    = 1           ! output interval [steps] for velocity
dn_Nu     = 1           ! output interval [steps] for Nusselt (torque)
dn_hdf5   = 100         ! output interval [steps] for HDF5 output
dn_prbs   = 1           ! output interval [steps] for time series data at probe locations
print_time_screen = 10  ! output interval [steps] for timestep info to stdout
/

&parameters_control
restart = 0          ! 0 = new run, 1,2 = restart from checkpoint
runtime = 7200       ! maximum WCT [s] for the job
/

&parameters_initialcondition
ic_tcbf    = T                ! Set Taylor-Couette base flow (T) or resting fluid (F), only when restart = 0
ic_pert    = T                ! Add perturbation on top of base flow (T) or not (F), only when restart = 0
ic_p(1, :) = 2.2d-2, 0, 2     ! 1st perturbation: amplitude and wavevector (a1, k_th1, k_z1)
ic_p(2, :) = 1.0d-2, 3, 0     ! 2nd perturbation: amplitude and wavevector (a2, k_th2, k_z2)
ic_p(3, :) = 0.0d-2, 0, 0     ! 3rd perturbation: amplitude and wavevector (a3, k_th3, k_z3)
ic_p(4, :) = 0.0d-2, 0, 0     ! 4th perturbation: amplitude and wavevector (a4, k_th4, k_z4)
ic_p(5, :) = 0.0d-2, 0, 0     ! 5th perturbation: amplitude and wavevector (a5, k_th5, k_z5)
ic_p(6, :) = 0.0d-2, 0, 0     ! 6th perturbation: amplitude and wavevector (a6, k_th6, k_z6)
/
