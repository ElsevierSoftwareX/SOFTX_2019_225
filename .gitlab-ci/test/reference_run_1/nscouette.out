!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
! NSCouette, a HPC code for DNS of Taylor-Couette flow                 !
!                                                                      !
! Copyright (C) 2016 Marc Avila, Bjoern Hof, Jose Manuel Lopez,        !
!                    Markus Rampp, Liang Shi                           !
!                                                                      !
! NSCouette is free software: you can redistribute it and/or modify    !
! it under the terms of the GNU General Public License as published by !
! the Free Software Foundation, either version 3 of the License, or    !
! (at your option) any later version.                                  !
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

starting NSCouette (b7439aed) using    4 MPI tasks,    2 OpenMP threads/task

   TASKS:   4 THREADS: 2    THIS:   0     Host:buildtest- Cores:   0   1
   TASKS:   4 THREADS: 2    THIS:   1     Host:buildtest- Cores:   2   3
   TASKS:   4 THREADS: 2    THIS:   2     Host:buildtest- Cores:   4   5
   TASKS:   4 THREADS: 2    THIS:   3     Host:buildtest- Cores:   6   7
  step=          10  dt=  8.648729775507614E-007
  step=          20  dt=  8.648729775507614E-007
  step=          30  dt=  8.648729775507614E-007
  step=          40  dt=  8.648729775507614E-007
  step=          50  dt=  8.648729775507614E-007
  step=          60  dt=  8.648729775507614E-007
  step=          70  dt=  8.648729775507614E-007
  step=          80  dt=  8.648729775507614E-007
  step=          90  dt=  8.648729775507614E-007
  step=         100  dt=  8.648729775507614E-007
 written coeff file to disk: coeff_REF_1.00000100                    
T0: number of timesteps              :   100
T0: total time incl I/O [s]:   1209.53
T0:  av time per timestep excl I/O [s]:     11.73
ALL: min/max av time per timestep [s]:   11.7268   11.7277
 /= Group                                              [s]     fraction
 |                                            ============ ============
 |_ [Root] (running)                           1209.646260        1.000
   |_ (own)                                         0.000242        0.000
   |_ main                                       1209.646011        1.000
     |_ (own)                                         0.947395        0.001
     |_ pretstp                                       3.442325        0.003
     |_ tstep                                      1174.124105        0.971
       |_ (own)                                        16.548435        0.014
       |_ nonlin  (200x)                              980.540529        0.835
         |_ (own)                                         0.231458        0.000
         |_ matvec  (200x)                               26.551012        0.027
         |_ tranf  (200x)                                52.727057        0.054
         |_ deriv  (200x)                               458.228843        0.467
         |_ fft  (200x)                                 299.040257        0.305
         |_ tranb  (200x)                               143.761902        0.147
       |_ pretstp  (2x)                                 3.493960        0.003
       |_ predic  (100x)                               83.726411        0.071
       |_ correc  (100x)                               88.377256        0.075
         |_ (own)                                        82.793368        0.937
         |_ mcorr  (100x)                                 5.583888        0.063
       |_ io energy  (10x)                              1.437514        0.001
     |_ io coeff                                     31.132186        0.026
